exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggTemplate.java

```diff
@@ -306,14 +685,29 @@ public AggOutcome doWork() {
 
           while (true) {
             // Cleanup the previous batch since we are done processing it.
+            long pre = allocator.getAllocatedMemory();
             for (VectorWrapper<?> v : incoming) {
               v.getValueVector().clear();
             }
+            long beforeAlloc = allocator.getAllocatedMemory();
+
```

Comment:
The doWork() function's basic outline is the same as before; this code change just peppered few lines of code in few places. I was trying to minimize the change ....